src/setup-beam.js

```diff
@@ -478,6 +483,28 @@ function isKnownBranch(ver) {
   return ['main', 'master', 'maint'].includes(ver)
 }
 
+function githubARMRunnerArchs() {
+  return ['ARM', 'ARM64']
+}
+
+function githubAMDRunnerArchs() {
+  return ['X86', 'X64']
+}
+
+function getRunnerOSArchitecture() {
+  // These options come from:
+  // https://docs.github.com/en/actions/learn-github-actions/variables#default-environment-variables
+  if (githubARMRunnerArchs().includes(process.env.RUNNER_ARCH)) {
+    return 'arm64'
+  }
+
+  if (githubAMDRunnerArchs().includes(process.env.RUNNER_ARCH)) {
+    return 'amd64'
+  }
+
+  return ''
```

Comment:
We should probably error out here or at least warn and default to `amd64` like we did before. We want to stop using the `/builds/otp/${osVersion}/builds.txt` URL without arch in the path eventually.

Reply:
Yeah, I'd prefer to error out, too, to make it explicit there was a bad choice in the process, somewhere.

Reply:
This has now been implemented as suggested.